develop/tutorials/articles/03-portlets/packaging-a-jsf-application.markdown

```diff
@@ -198,18 +191,22 @@ approaches can be done by using the JSF Expression Language (EL):
         this.userService = userService;
     }
 
-Then you can write an
+Or you could more directly use the facesContext with the Liferay Faces Bridge as follows:
+
+	UserLocalService userLocalService = (UserLocalService) facesContext.getExternalContext()
+						.getApplicationMap().get("userLocalService");
+
+This approach above, using the `@Reference` annoation, allows the portlet to enjoy the full OSGi lifecycle, so that the portlet will not be available for use if the required service is not available.
+
+Alternatively, you could write an
 [ELResolver](http://docs.oracle.com/javaee/7/api/javax/el/ELResolver.html) that
 does the following:
 
     BundleContext bundleContext = FrameworkUtil.getBundle(getClass()).getBundleContext();
     ServiceReference<?>[] refs = ctx.getServiceReferences(UserService.class.getName(), null);
     userService = (UserService) bundleContext.getService(refs[0]);
 
-The drawback of this technique is that you don't enjoy the full OSGi lifecycle.
-In other words, the WAB will be placed in-service whether or not its dependent
-services are available. Additionally, it will not go out-of-service when any
-dependent services become unavailable.
+The drawback of this technique to get the service is that you don't enjoy the full OSGi lifecycle.  In other words, the WAB will be placed in-service whether or not its dependent services are available. Additionally, it will not go out-of-service when any dependent services become unavailable.
```

Comment:
"it (the OSGi module) will not go out-of-service" should be replaced with "the portlet annotated with @Component will not be deactivated"